### PR TITLE
docs: Validate RFCs filenames

### PR DESCRIPTION
**Pull request by frondeus**
This PR adds RFC filename validation!

No more invalid RFC names or conflicts between ordering number.
I chose `cargo xtask` way, although it means on CI it needs to be compiled. Perhaps we could cache it somehow but I'm not an expert here.

To run it manually: `cargo xtask check-docs`

### CODE REVIEW

**Comment by frondeus** [comment]
I also fixed one RFC - I detected it thanks to this feature :)

**Review by Kononnable — APPROVED** [review]
Suggestions are optional - should work properly without adding them.

File: .github/workflows/mdbook-check.yml [hunk 1]
```diff
@@ -5,6 +5,46 @@ on:
...
+          override: true
+          components: clippy, rustfmt
+
+      - name: Cache cargo directories
+        uses: actions/cache@v2.1.6
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
```
[the remark this answers is not in the document]
Comment (frondeus) [reply .github/workflows/mdbook-check.yml:28]: Okay, I agree. Now do you mean to change it to:
```suggestion
          key: ${{ runner.os }}-cargo-${{ hashFiles('xtask/Cargo.toml') }}
```
?

File: .github/workflows/mdbook-check.yml [hunk 2]
```diff
@@ -5,6 +5,46 @@ on:
...
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Cache cargo target dir
+        uses: actions/cache@v2.1.6
+        with:
+          path: target
+          key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
```
[the remark this answers is not in the document]
Comment (frondeus) [reply .github/workflows/mdbook-check.yml:35]:
```suggestion
          key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('xtask/Cargo.toml') }}
```

**Review by Kononnable — APPROVED** [review]

**Pull request merged**